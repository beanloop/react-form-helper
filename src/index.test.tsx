import * as React from 'react'
import {snap} from 'tscomp'
import {FormHelper} from './index'

describe('FormHelper', () => {
  it('should render the passed fields', () => {
    snap(
      <FormHelper
        value={{a: 'a'}}
        onSave={() => null}
        fields={[
          {path: ['a']},
          {path: ['b']},
        ]}
      />
    )
  })

  it('should render using specified components', () => {
    snap(
      <FormHelper
        inputComponent='some-input'
        value={{a: 'a', c: 'c'}}
        onSave={() => null}
        fields={[
          {path: ['a']},
          {path: ['b']},
          {component: 'other-input', path: ['c']},
          {component: 'other-input', path: ['d']},
        ]}
      />
    )
  })

  it('should not pass errors to the browser inputs', () => {
    snap(
      <FormHelper
        value={{}}
        onSave={() => null}
        fields={[
          {path: ['a'], error: 'a error'},
          {path: ['b'], validationError: 'b', validations: {b: {text: 'b error'}}},
          {path: ['c'], validations: {c: {text: 'c error', validation: () => false}}},
        ]}
      />
    )
  })

  it('should pass errors to custom inputs', () => {
    snap(
      <FormHelper
        inputComponent='input'
        value={{c: 'value'}}
        onSave={() => null}
        fields={[
          {path: ['a'], error: 'a error'},
          {path: ['b'], validationError: 'b', validations: {b: {text: 'b error'}}},
          {path: ['c'], validations: {c: {text: 'c error', validation: () => false}}},
        ]}
      />
    )
  })

  it('should not run custom validators on no value', () => {
    snap(
      <FormHelper
        inputComponent='input'
        value={{}}
        onSave={() => null}
        fields={[
          {path: ['a'], validations: {a: {text: 'a error', validation: () => {throw 'should not be called'}}}},
        ]}
      />
    )
  })

  it('should pass errors to untouched inputs if errorOnTouched is true', () => {
    snap(
      <FormHelper
        errorOnTouched
        inputComponent='input'
        value={{c: 'value'}}
        onSave={() => null}
        fields={[
          {path: ['a'], error: 'a error'},
          {path: ['b'], validationError: 'b', validations: {b: {text: 'b error'}}},
          {path: ['c'], validations: {c: {text: 'c error', validation: () => false}}},
        ]}
      />
    )
  })
})
