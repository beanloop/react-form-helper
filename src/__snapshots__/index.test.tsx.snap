exports[`FormHelper should not pass errors to the browser inputs 1`] = `
<form
  id={undefined}
  onSubmit={[Function]}
  style={undefined}>
  <input
    disabled={undefined}
    onChange={[Function]}
    value="" />
  <input
    disabled={undefined}
    onChange={[Function]}
    value="" />
  <input
    disabled={undefined}
    onChange={[Function]}
    value="" />
</form>
`;

exports[`FormHelper should not run custom validators on no value 1`] = `
<form
  id={undefined}
  onSubmit={[Function]}
  style={undefined}>
  <input
    disabled={undefined}
    onChange={[Function]}
    value="" />
</form>
`;

exports[`FormHelper should pass errors to custom inputs 1`] = `
<form
  id={undefined}
  onSubmit={[Function]}
  style={undefined}>
  <input
    disabled={undefined}
    error="a error"
    onChange={[Function]}
    value="" />
  <input
    disabled={undefined}
    error="b error"
    onChange={[Function]}
    value="" />
  <input
    disabled={undefined}
    error="c error"
    onChange={[Function]}
    value="value" />
</form>
`;

exports[`FormHelper should pass errors to untouched inputs if errorOnTouched is true 1`] = `
<form
  id={undefined}
  onSubmit={[Function]}
  style={undefined}>
  <input
    disabled={undefined}
    error={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    value="" />
  <input
    disabled={undefined}
    error={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    value="" />
  <input
    disabled={undefined}
    error={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    value="value" />
</form>
`;

exports[`FormHelper should render the passed fields 1`] = `
<form
  id={undefined}
  onSubmit={[Function]}
  style={undefined}>
  <input
    disabled={undefined}
    onChange={[Function]}
    value="a" />
  <input
    disabled={undefined}
    onChange={[Function]}
    value="" />
</form>
`;

exports[`FormHelper should render using specified components 1`] = `
<form
  id={undefined}
  onSubmit={[Function]}
  style={undefined}>
  <some-input
    disabled={undefined}
    onChange={[Function]}
    value="a" />
  <some-input
    disabled={undefined}
    onChange={[Function]}
    value="" />
  <other-input
    disabled={undefined}
    onChange={[Function]}
    value="c" />
  <other-input
    disabled={undefined}
    onChange={[Function]}
    value="" />
</form>
`;
